simapp/app.go

```diff
@@ -289,7 +289,10 @@ func NewSimApp(
 	// NOTE: Any module instantiated in the module manager that is later modified
 	// must be passed by reference here.
 	app.mm = module.NewManager(
-		genutil.NewAppModule(app.AccountKeeper, app.StakingKeeper, app.BaseApp.DeliverTx),
+		genutil.NewAppModule(
+			app.AccountKeeper, app.StakingKeeper, app.BaseApp.DeliverTx,
+			AminoJSONTxDecoder(encodingConfig), encodingConfig.TxGenerator.TxEncoder(),
```

Comment:
I think I don't understand why we need an explicit JsonTxDecoder here? from what I understood with TxGenerator, TxGenerator already takes care of encoding/decoding.